addon/services/redux.js

```diff
@@ -0,0 +1,53 @@
+import Ember from 'ember';
+import redux from 'redux';
+import reducers from '../reducers/index';
+import enhancers from '../enhancers/index';
+import middlewareConfig from '../middleware/index';
+
+const { assert, isArray, get } = Ember;
+
+// Handle "classic" middleware exports (i.e. an array), as well as the hash option
+const extractMiddlewareConfig = (mc) => {
+  assert(
+    'Middleware must either be an array, or a hash containing a `middleware` property',
+    isArray(mc) || mc.middleware
+  );
+  return isArray(mc) ? { middleware: mc } : mc;
+}
+
+const { createStore, applyMiddleware, combineReducers, compose } = redux;
+
+const makeStoreInstance = ({middlewareConfig, reducers, enhancers}) => {
+  const { middleware, setup = () => {} } = extractMiddlewareConfig(middlewareConfig);
+  const createStoreWithMiddleware = compose(applyMiddleware(...middleware), enhancers)(createStore);
+  const reducer = typeof reducers === 'function' ? reducers : combineReducers(reducers);
+  const store = createStoreWithMiddleware(reducer);
+  setup(store);
+  return store;
+};
+
+export default Ember.Service.extend({
+  middlewareConfig,
```

Comment:
Is there a reason this field is named `middlewareConfig` rather than just `middleware` (matching the pattern of `reducers`, `enhancers`)? Just want to be careful with the naming since these will become part of the public API when overriding the default implementation here.